tests/testTez.ml

```diff
@@ -1,6 +1,10 @@
 open OUnit2
 open TestLib
 
+(* TODO: Would be nice to have randomized tests, and even check the bounds
+ * tightly. Perhaps we should consider doing something similar for the other
+ * types (Lqt, Kit, Ctez). *)
+
```

Comment:
I don't really know how to implement this, but checking those implementations against `ligo` would be pretty nice too. eg. whether running `mul_tez_nat` on Ocaml side always give some results as `*` in ligo (specialised to the same type).

Reply:
Ah, yeah, that'd be great indeed. If we could have checked against Michelson we'd have caught this misrepresentation issue ages ago too :sweat_smile: I'm not entirely sure either, but I'll leave the `TODO` comment there, until we can figure something out. Alternatively we could open an issue, though it doesn't feel very pressing at the moment I guess.